platform/admin/configurator/admin-configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/ResultImpl.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator.impl;
+
+import java.util.Optional;
+
+import org.codice.ddf.admin.configurator.Result;
+import org.codice.ddf.admin.configurator.Status;
+
+public class ResultImpl implements Result {
+
+    private final Status status;
+
+    private final Throwable badOutcome;
+
+    private final String configId;
```

Comment:
Config ID seems to only be related to managed services.  Can we extract operation specific data into something more generic?  The cop out would be a map but I could also see something like Optional< T > data where T would be a generic for some kind of operation specific result data type.  For example I mentioned the CreatedManagedServicePid type.